sensors/src/main/scala/com/lynbrookrobotics/potassium/sensors/position/xyPositionCalculator.scala

```diff
@@ -0,0 +1,39 @@
+package com.lynbrookrobotics.potassium.sensors.position
+
+import com.lynbrookrobotics.potassium.units.{Point, Value3D}
+import com.lynbrookrobotics.potassium.{PeriodicSignal, Signal}
+import squants.Angle
+import squants.space._
+import squants.time.{Seconds, Time}
+
+/**
+  * Calculates the current xy position given the current distance traveled since
+  * start of calculations and the current angle
+  * @param initPosition initial position
+  * @param angle trigonometric angle, where 0 is defined at the x axis
+  * @param totalDistanceTraveled signal for total distance traveled
+  */
+class xyPositionCalculator(initPosition: Point,
+                           angle: PeriodicSignal[Angle],
+                           totalDistanceTraveled: PeriodicSignal[Length]) {
+  def this(angle: PeriodicSignal[Angle],
+           totalDistanceTraveled: PeriodicSignal[Length]) = {
+    this(new Point(Feet(0), Feet(0)), angle, totalDistanceTraveled)
+  }
+
+  private val initAngle = angle.currentValue(Seconds(0))
```

Comment:
I need the previously computed value of the angle to do my computations, for which angle has to be a PeriodicSignal. In addition, converting from a PeriodicSignal that the angle will be provided with from the gyro to a Signal will be a painful